addons/mail/models/res_users.py

```diff
@@ -94,9 +95,26 @@ def _init_messaging(self):
             'public_partners': list(self.env.ref('base.group_public').sudo().with_context(active_test=False).users.partner_id.mail_partner_format().values()),
             'shortcodes': self.env['mail.shortcode'].sudo().search_read([], ['source', 'substitution', 'description']),
             'starred_counter': self.partner_id._get_starred_count(),
+        } if not 'guest' in self.env.context else {
```

Comment:
if there is nothing common, would be more readable in a traditional if/else multiline block